pkg/apis/pingcap/v1alpha1/defaulting/dmcluster.go

```diff
@@ -55,6 +56,9 @@ func setMasterSpecDefault(dc *v1alpha1.DMCluster) {
 			dc.Spec.Master.BaseImage = defaultMasterImage
 		}
 	}
+	if dc.Spec.Master.MaxFailoverCount == nil {
+		dc.Spec.Master.MaxFailoverCount = pointer.Int32Ptr(3)
+	}
```

Comment:
I think set a default `MaxFailoverCount` value is better. Some users may switch on `auto-failover` option in `tidb-operator` but forget about this argument and will find that failover doesn't work.